packages/components/src/components/popup/popup.tsx

```diff
@@ -28,10 +27,10 @@ const Popup = ({
     tabs_detail,
     title,
     togglePopupModal,
-}) => {
-    const [overlay_ref, setOverlayRef] = React.useState(null);
-    const [is_overlay_shown, setIsOverlayShown] = React.useState(false);
-    const context_value = {
+}: TPopupContext) => {
+    const [overlay_ref, setOverlayRef] = React.useState<HTMLDivElement | null>();
+    const [is_overlay_shown, setIsOverlayShown] = React.useState<boolean>(false);
```

Comment:
You don't need to define types for these kind of states because it takes it's default value type by default so you can remove it from here!
```suggestion
    const [is_overlay_shown, setIsOverlayShown] = React.useState(false);
```